### ISSUE TITLE
minio / s3 media backend

### ISSUE THREAD

**Issue by v411e** [silverbulletmd/silverbullet#519]
I use silverbullet together with the great [ghost-plug](https://github.com/silverbulletmd/silverbullet-ghost) to publish my blog posts. This is very cool and I love this workflow. However, adding images is not that easy. When uploading them to silverbullet, they are not accessible without authentication. Normally, this is exactly what you want. But when writing a blog post, I want the images to be accessible for everyone on the internet.

Currently, I upload the images to a selfhosted PicoShare instance and add them to my post via link. This is a bit hacky and not very convenient.

[HedgeDoc](https://hedgedoc.org/) has a feature called ["Media Backend"](https://docs.hedgedoc.org/guides/minio-image-upload/) which is about exactly what I would wish to have in silverbullet. Pictures are directly uploaded into a minio/s3 bucket. The resulting link with a random, long ID would be available for anyone that knows the exact link and also for those visiting my ghost blog.

What do you think about such a feature?

**Comment by zefhemel** [comment]
Yeah, I never got around to implement the image upload part of the Ghost plug, probably because I don't use images enough myself. An alternative solution here is to actually have the Ghost plug support images and just upload them via the Ghost API which I'm sure is possible. I'd have to look into the details.

**Comment by v411e** [comment]
Oh yes, that's maybe even better :) I might also have a look into the API.
